docs/book/src/tasks/experimental-features/machineset-preflight-checks.md

```diff
@@ -0,0 +1,12 @@
+# Experimental Feature: MachineSetPreflightChecks (alpha)
+
+The `MachineSetPreflightChecks` feature is introduced to provide additional safety while creating new Machines for a MachineSet.
+
+**Feature gate name**: `MachineSetPreflightChecks`
+
+**Variable name to enable/disable the feature gate**: `EXP_MACHINE_SET_PREFLIGHT_CHECKS`
+
+The following preflight checks are performed when the feature is enabled:
+* ControlPlaneStable
+* KubeadmVersion
+* KubernetesVersion
```

Comment:
Is it possible to provide a user-oriented explanation about what is check is implemented/why it is important?

Also, what about adding a note about the possibility to skip preflight check by exception/on a MD or MS bases once the feature flag is enabled?

Reply:
I assume the plan was to do this in https://github.com/kubernetes-sigs/cluster-api/issues/8717 as a follow-up

Reply:
Yup, there is a follow-up issue to improve the documentation as discussed here: https://github.com/kubernetes-sigs/cluster-api/pull/8595#discussion_r1200422174